apps/src/gamelab/AnimationTab/ItemLoopToggle.jsx

```diff
@@ -0,0 +1,59 @@
+/** @file controls below an animation looping toggle */
+'use strict';
+
+var React = require('react');
+var color = require('../../color');
+var Radium = require('radium');
+import * as PropTypes from '../PropTypes';
+
+var styles = {
+  loopToggleStyle: {
+    width: 30,
+    height: 30,
+    backgroundColor: color.purple,
+    borderRadius: '50%',
+    padding: 2,
+    cursor: 'pointer'
+  },
+  loopIconStyle: {
+    paddingTop: 3,
+    paddingLeft: 3,
+    position: 'relative'
+  }
+};
+
+/**
+ * The toggle that controls whether the animation loops frames.
+ */
+
+const ItemLoopToggle = React.createClass({
+  getDefaultProps: function () {
+    return {
+      looping: true
+    };
+  },
+
+  propTypes: {
+    style: React.PropTypes.object,
+    onToggleChange: React.PropTypes.func,
+    looping: React.PropTypes.bool
+  },
+
+  toggleClicked() {
+    this.props.onToggleChange(!this.props.looping);
+  },
+
+  render() {
+    let iconImageSrc = "/blockly/media/gamelab/looping-one-time.png";
+    if (this.props.looping === true) {
+      iconImageSrc = "/blockly/media/gamelab/looping-continuous.png";
+    }
```

Comment:
This would express nicely as a ternary, and you could optionally even deduplicate part of the string.  Personally I'd do one of these:

```javascript
const iconImageName = this.props.looping ? 'looping-continuous' : 'looping-one-time';
const iconImageSrc = `/blockly/media/gamelab/${iconImageName}.png`;

// or

const iconImageSrc = this.props.looping ?
    '/blockly/media/gamelab/looping-continuous.png' :
    '/blockly/media/gamelab/looping-one-time.png';
```

To me, they sort of make a stronger assertion that nothing more complicated than picking between two things is going on, whereas the existing `let x = 'a'; if (n) x = 'b';` might let someone be overly clever and put other behavior in the if-statement down the road.

But maybe I'm just super-paranoid :grin: